extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/client/RedisClientRecorder.java

```diff
@@ -92,13 +94,38 @@ public ConfigurationException get() {
                                         "You must at least configure `quarkus.redis." + name + ".hosts`.");
                             }
                         });
-                clients.computeIfAbsent(name,
+                RedisClientAndApi redisClientAndApi = clients.computeIfAbsent(name,
                         x -> new RedisClientAndApi(name, VertxRedisClientFactory.create(name, vertx, actualConfig, tlsRegistry),
                                 metrics));
+
+                if (actualConfig.configureClientName()) {
+                    String clientName = actualConfig.clientName().orElse(name);
+                    redisClientAndApi.redis.send(Request.cmd(Command.CLIENT, new Object[] {
+                            "SETNAME", clientName
+                    })).subscribe()
```

Comment:
This approach will lead to race conditions. The execution of the command is async and thus, someone using the client immediately to emit another command may race with this (and may not have the client set).

This approach would work but we would need to make sure 
1. Redis is available
2. Every other commands are enqueued until this one terminate successfully